core/src/main/java/tech/pegasys/web3signer/core/Eth1Runner.java

```diff
@@ -93,7 +97,12 @@ protected void populateRouter(final Context context) {
                 false))
         .failureHandler(errorHandler);
 
-    addReloadHandler(router, signerProvider, context.getErrorHandler());
+    final ArtifactSignerProvider secpArtifactSignerProvider =
+        new SecpArtifactSignerProviderAdpater(signerProvider);
+
+    // The order of the elements in the list DO matter
+    addReloadHandler(
+        router, List.of(signerProvider, secpArtifactSignerProvider), context.getErrorHandler());
```

Comment:
nit: if the order matters maybe can express that in the type by using ArrayList explicitly (and maybe extracting a variable with a name that mentions 'ordered')

Reply:
Lists are by definition ordered. Also rather not tie this to a specific implementation of List as it doesn't matter for the reloadHandler